datasets/DEFRA-ENV01-Emissions-of-air-pollutants/main.py

```diff
@@ -0,0 +1,203 @@
+        observations = sector.shift(RIGHT).fill(RIGHT).is_not_blank()
+        
+        dimensions = [
+                    HDim(sector, 'Sector', 'DIRECTLY', LEFT),
+                    HDim(year, 'Period', 'DIRECTLY', ABOVE),
+                    HDimConst('Measure Type', 'emissions'),
+                    HDimConst('Unit', "tonnes")
+                ]
+        
+        tidy_sheet = ConversionSegment(tab, dimensions, observations)
+        table = tidy_sheet.topandas()
+        table['OBS'] = table['OBS']*1000
+        trace.store('dataframe' + tab.name + str(j), table)
+        
+        df_67 = trace.combine_and_trace(title, 'dataframe' + tab.name + str(j)).fillna('')
+        df_67['Period'] = df_67['Period'].str.replace('\.0', '')
+        df_67.rename(columns={'OBS':'Value'}, inplace=True)
+        df_67['Value'] = df_67['Value'].fillna('-1')
+        df_67['Value'] = pd.to_numeric(df_67['Value'], downcast='float')
+
+        df_67['Value'] = (df_67['Value'].fillna(0).astype(float).round().astype(int).astype(object).where(df_67['Value'].notnull()))
+       
+        df_67['Total Value'] = ' ' 
+        df_67.loc[(df_67['Sector'] == 'NATIONAL TOTAL'), 'Total Value'] = df_67.loc[(df_67['Sector'] == 'NATIONAL TOTAL'), 'Value']
+        df_67.loc[(df_67['Sector'] == 'NATIONAL TOTAL'), 'Value'] = ' '
+        df_67.loc[(df_67['Sector'] == 'NATIONAL TOTAL'), 'Sector'] = 'All'
+        df_67 = df_67[['Sector', 'Period', 'Measure Type', 'Value', 'Total Value', 'Unit']]
+        j += 1
+        cubes.add_cube(metadata, df_67, metadata.dataset.title)
+
+
+        title2 = title_list[j]
+        metadata.dataset.title = title2
+        print(tab.name, ' ', title2)
+        columns2 = ['Sector', 'Period', 'Measure Type', 'Value']
+        trace.start(title2, tab, columns2, distribution.downloadURL)
+        measures = tab.excel_ref('C').filter("% of total emissions").expand(RIGHT).is_not_blank()
+        year2 = measures.shift(UP).shift(UP)
+        sector2 = tab.excel_ref('C').filter('% of total emissions').shift(-2,0).fill(DOWN).is_not_blank()
+        observations2 = sector2.shift(RIGHT).fill(RIGHT).is_not_blank()
+        
+        dimensions2 = [
+                    HDim(sector2, 'Sector', 'DIRECTLY', LEFT),
+                    HDim(year2, 'Period', 'DIRECTLY', ABOVE),
+                    HDimConst('Measure Type', "% of total emissions")
+            ]
+        tidi_sheet_2 = ConversionSegment(tab, dimensions2, observations2)
+        table2 = tidi_sheet_2.topandas()
+        trace.store('dataframe' + tab.name + str(j), table2)
+        df_percent = trace.combine_and_trace(title2, 'dataframe' + tab.name + str(j)).fillna('')
+        df_percent['Period'] = df_percent['Period'].str.replace('\.0', '')
+        df_percent.rename(columns={'OBS':'Value'}, inplace=True)
+        df_percent['Value'] = pd.to_numeric(df_percent['Value'], downcast='float')
+        df_percent['Value'] = df_percent['Value'].round(2)
```

Comment:
Should we be rounding the percentage or values in general? Something to maybe ask DM